Jenkins-Repro-Steps.md

```diff
@@ -0,0 +1,162 @@
+**Only trigger these builds if there is an issue revealed in the dailies that does not repro in the
+rolling builds (to save build machine resources).**
+
+### Windows (Daily)
+
+Repro with these commands:
+
+```
+jenkins\buildone.cmd x64 debug
+jenkins\testone.cmd x64 debug -includeSlow
+```
+
+When testing a fix please request:
+
+```
+@dotnet-bot test slow tests please
+```
+
+### Linux (Daily)
+
+Repro with these commands:
+
+```
+./build.sh --debug
+./test/runtests.sh -d
+```
+
+When testing a fix please request:
+
+```
+@dotnet-bot test linux tests please
+```
+
+## DisableJIT Builds (Windows-only)
+
+Repro with these commands:
+
+```
+jenkins\buildone.cmd x64 debug "/p:BuildJIT=false"
+jenkins\testone.cmd x64 debug -disablejit
+```
+
+When testing a fix please request:
+
+```
+@dotnet-bot test nojit tests please
+```
+
+(You can also replace `test nojit tests` with `test disablejit tests` if you prefer -- they are equivalent.)
+
+## Legacy Builds (Windows-only)
+
+_Note: **These tests are designed to run under VS 2013 + Windows 7.**
+In most cases, failures are related to running under msbuild 12.0 so you can most likely
+reproduce failures using msbuild 12.0 without needing to run on Windows 7. If you still cannot repro,
+you might consider setting up a Windows 7 VM to repro._
+
+Repro with these commands:
+
+```
+jenkins\buildone.cmd x64 debug msbuild12
+jenkins\testone.cmd x64 debug -win7 -includeSlow
+```
+
+When testing a fix please request the following:
+
+```
+@dotnet-bot test dev12 tests please
+```
+
+(You can also replace `test dev12 tests` with `test legacy tests` if you prefer -- they are equivalent.)
+
+# Footnotes
+
+## Windows Alternate Flags
+
+You can replace `x64` with `x86` or `arm` (running tests on `arm` builds is not supported).
+
+You can replace `debug` with `test` or `release` (running tests on `release` builds is not supported).
+
+## Linux Alternate Flags
+
+See [Building ChakraCore](https://github.com/Microsoft/ChakraCore/wiki/Building-ChakraCore#linux)
+
+You can replace `--debug` with `--test-build`.
+
+You can replace `-d` (corresponding with `--debug` in `build.sh`)
+with `-t` (corresponding with `--test-build` in `build.sh`).
+
+## Systems
+
+See [Building ChakraCore](https://github.com/Microsoft/ChakraCore/wiki/Building-ChakraCore)
+for specific details on the platforms and build configurations we support and requirements to build them.
+
+* **Windows:** We will primarily support Windows 8.1+ with VS 2015.
+We currently support systems with Windows 7 or VS 2013 as Legacy configurations.
+* **Linux:** We will support Ubuntu 16.04 LTS with Clang 3.8+.
+* **OS X:** We will support OS X with Clang 3.8+.
+
+## Shorthand
+
+For purposes of shorthand the following terms are considered equivalent here:
+
+* VS 2015 = Visual Studio 2015 = Dev14 = msbuild 14.0
+* VS 2013 = Visual Studio 2013 = Dev12 = msbuild 12.0
+* Windows 10 = [for our purposes, no important differences from] Windows 8.1 =
```

Comment:
Windows 10 = Windows 8.1 is going to cause some confusion. I'd prefer we use something like Windows 8.1+ or Windows 8.1 and beyond to avoid confusion.